x-pack/plugins/metrics_data_access/server/lib/metrics/index.ts

```diff
@@ -39,6 +39,15 @@ export const query = async (
     },
   };
   const hasGroupBy = Array.isArray(options.groupBy) && options.groupBy.length > 0;
+  const groupInstanceFilter: Array<Record<string, any>> = [];
+  options.groupInstance?.map((group, index) => {
+    const key = options.groupBy?.[index];
+    if (typeof key === 'string' && group) {
+      groupInstanceFilter.push({
+        term: { [key]: group },
+      });
+    }
+  });
```

Comment:
```suggestion
    const groupInstanceFilter = options.groupInstance?.map((group, index) => {
    const key = options.groupBy?.[index];
    if (typeof key === 'string' && group) {
     return {
        term: { [key]: group },
      };
    }
  });
```

Reply:
@fkanout @cauemarcondes This suggestion commit breaks the build. When I was writing this logic, I tried not to introduce explicit type but typescript cannot guess the type correctly as you see in the following screenshot. So I will revert it for now, and if you know a way to fix this wrong type, please let me know.

![image](https://github.com/elastic/kibana/assets/12370520/1f5f5a2f-29e4-4c82-a44d-3382462b3bf6)

Reply:
Yeah, the error is expected actually. As you map through the array you add the term or undefined. 

You could solve this by using reduce:
```
const groupInstanceFilter = options.groupInstance.reduce<Array<Record<string, any>>>(
    (acc, group, index) => {
      const key = options.groupBy?.[index];
      if (typeof key === 'string' && group) {
        acc.push({ term: { [key]: group } });
      }
      return acc;
    },
    []
  );
```

But, it's just a suggestion.

Reply:
Ah, right! Thanks for the suggestion, I will give it a try.

Reply:
Updated in [6c7af97](https://github.com/elastic/kibana/pull/181689/commits/6c7af9762b45c27fa364b2d4395c507f224c6c30)